docs/api/portable-stories-jest.md

```diff
@@ -0,0 +1,288 @@
+| play       | `(context) => Promise<void> \| undefined` | Executes the [play function](#4-play-optional) of a given story |
+
+## composeStory
+
+You can use `composeStory` if you wish to compose a single story for a component.
+
+<!-- prettier-ignore-start -->
+
+<CodeSnippets
+  paths={[
+    'react/portable-stories-jest-compose-story.ts.mdx',
+    'vue/portable-stories-jest-compose-story.ts.mdx',
+  ]}
+/>
+
+<!-- prettier-ignore-end -->
+
+### Type
+
+<!-- prettier-ignore-start -->
+```ts
+(
+  story: Story export,
+  componentAnnotations: Meta,
+  projectAnnotations?: ProjectAnnotations,
+  exportsName?: string
+) => ComposedStoryFn
+```
+<!-- prettier-ignore-end -->
+
+### Parameters
+
+#### `story`
+
+(**Required**)
+
+Type: `Story export`
+
+Specifies which story you want to compose.
+
+#### `componentAnnotations`
+
+(**Required**)
+
+Type: `Meta`
+
+The default export from the stories file containing the [`story`](#story).
+
+#### `projectAnnotations`
+
+Type: `ProjectAnnotation | ProjectAnnotation[]`
+
+Specifies the project annotations to be applied to the composed story.
+
+This parameter is provided for convenience. You should likely use [`setProjectAnnotations`](#setprojectannotations) instead. Details about the `ProjectAnnotation` type can be found in that function's [`projectAnnotations`](#projectannotations-2) parameter.
+
+This parameter can be used to [override](#overriding-globals) the project annotations applied via `setProjectAnnotations`.
+
+#### `exportsName`
+
+Type: `string`
+
+You probably don't need this. Because `composeStory` accepts a single story, it does not have access to the name of that story's export in the file (like `composeStories` does). If you must ensure unique story names in your tests and you cannot use `composeStories`, you can pass the name of the story's export here.
+
+### Return
+
+Type: `ComposedStoryFn`
+
+A single [composed story](#return).
+
+## setProjectAnnotations
+
+This API should be called once, before the tests run, typically in a [setup file](https://jestjs.io/docs/configuration#setupfiles-array). This will make sure that whenever `composeStories` or `composeStory` are called, the project annotations are taken into account as well.
+
+```ts
+// setup-portable-stories.ts
+// Replace <your-renderer> with your renderer, e.g. react, vue3
+import { setProjectAnnotations } from '@storybook/<your-renderer>';
+import * as addonAnnotations from 'my-addon/preview';
+import * as previewAnnotations from './.storybook/preview';
+
+setProjectAnnotations([previewAnnotations, addonAnnotations]);
+```
+
+<Callout variant="warning">
+
+Sometimes a story can require an addon's [decorator](../writing-stories/decorators.md) to render properly. For example, an addon can apply a decorator that wraps your story in the necessary router context. In this case, you must include that addon's `preview` export in the project annotations set.
```

Comment:
@yannbf — How's this? It's the same text for all three pages.

Reply:
The tricky part is that this only applies for the following scenario:
- Addons which provide decorators or loaders, but only internally, and those decorators are needed for your story to render properly.

For instance, addon-themes, msw addon, etc, all have decorators/loaders, however not in their preview preset, but exported instead (so the users have to declare them manually). In that case, these annotations will be applied correctly as they are part of the user's .storybook/preview file

Reply:
Revised:

<img width="683" alt="" src="https://github.com/storybookjs/storybook/assets/486540/3ad0bdcf-5d09-4f51-8283-7b08d4181b85">

Reply:
That's great!